extensions/control-plane/provision/provision-gcs/src/main/java/org/eclipse/edc/connector/provision/gcp/GcsResourceDefinition.java

```diff
@@ -15,29 +15,44 @@
 package org.eclipse.edc.connector.provision.gcp;
 
 import org.eclipse.edc.connector.transfer.spi.types.ResourceDefinition;
+import org.eclipse.edc.spi.types.domain.DataAddress;
 
 import java.util.Objects;
 
 public class GcsResourceDefinition extends ResourceDefinition {
 
+    private DataAddress dataAddress;
     private String location;
+
+    private String projectId;
     private String storageClass;
 
+
     private GcsResourceDefinition() {
     }
 
+    public DataAddress getDataAddress() {
```

Comment:
Since we are using dataAddress to access service account info in vault, we should update the generator to ensure that the field is also copied:
https://github.com/eclipse-edc/Connector/blob/b24a5cacbc9fcabdfd8020d779399b3e56856661/extensions/control-plane/provision/provision-gcs/src/main/java/org/eclipse/edc/connector/provision/gcp/GcsConsumerResourceDefinitionGenerator.java#L42